codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/validation/UnaryFunctionCall.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.aws.typescript.codegen.validation;
+
+import java.util.regex.Pattern;
+
+/**
+ * For handling expressions that may be unary function calls.
+ */
+class UnaryFunctionCall {
```

Comment:
I think this repo and smithy-ts have different static analysis requirements. It might've needed to be abstract there when there were no instance level methods or properties. That check made sense to me.